src/modules/avformat/link_avfilter.c

```diff
@@ -827,14 +827,17 @@ static int link_get_audio(mlt_frame frame,
 #endif
             || *samples != pdata->avoutframe->nb_samples
             || *frequency != pdata->avoutframe->sample_rate) {
-            mlt_log_error(self,
-                          "Unexpected return format c %d->%d\tf %d->%d\tf %d->%d\n",
-                          *channels,
+
+            int expected_channels;
 #if HAVE_FFMPEG_CH_LAYOUT
-                          pdata->avoutframe->ch_layout.nb_channels,
+            expected_channels = pdata->avoutframe->ch_layout.nb_channels;
```

Comment:
Move this up to line 822 so that you can remove the same #if up there.

Reply:
I prefer to move it up in scope. Otherwise, a future developer (me) will have to try to remember why we did it differently in two different places. Thanks for humoring me.